src/webview/MessageList.js

```diff
@@ -93,18 +91,20 @@ type SelectorProps = {|
   // The remaining props contain data specific to the particular narrow or
   // particular messages we're displaying.  Data that's independent of those
   // should go in `BackgroundData`, above.
-  initialScrollMessageId: number | null,
   fetching: Fetching,
-  messages: $ReadOnlyArray<Message | Outbox>,
   htmlPieceDescriptorsForShownMessages: HtmlPieceDescriptor[],
   typingUsers: $ReadOnlyArray<UserOrBot>,
 |};
 
 // TODO get a type for `connectActionSheet` so this gets fully type-checked.
 export type Props = $ReadOnly<{|
   narrow: Narrow,
+  messages: $ReadOnlyArray<Message | Outbox>,
+  initialScrollMessageId: number | null,
   showMessagePlaceholders: boolean,
   startEditMessage: (editMessage: EditMessage) => void,
+  disableFetching?: true,
```

Comment:
This name seems a bit misleading because it doesn't affect any logic that controls actually *fetching*. Rather it suppresses the bits of UI that indicate that we are fetching. Perhaps `suppressFetchingIndicator`, similar to the `suppressTypingIndicator` elsewhere in the branch?

Huh, or in fact: I think we could just never pass this prop, and we'd get the same behavior. We'd go look up the narrow in the fetching state, and not find it, and return...
```js
/** The value implicitly represented by a missing entry in FetchingState. */
export const DEFAULT_FETCHING = { older: false, newer: false };
```

That seems better. After all if we *did* start doing fetching when you scroll to the top of a search narrow, we'd want that to get indicated just the same way as in other narrows.